netket/machine/jax.py

```diff
@@ -347,6 +347,160 @@ def JaxRbm(hilbert, alpha, dtype=complex):
+                MPS tensors is constructed as a sequence of identical unit cells
+                consisting of symperiod tensors. if None, symperiod equals the
+                number of physical degrees of freedom.
+                default=None
+            dtype: complex or float, whether the variational parameters of the MPS
+                are real or complex. default=complex
+
+        returns:
+            Jax machine of the Matrix Product state.
+    """
+    return Jax(
+        hilbert,
+        stax.serial(MpsPeriodicLayer(hilbert, bond_dim, diag, symperiod, dtype)),
+        dtype=dtype,
+    )
+
+
+def MpsPeriodicLayer(hilbert, bond_dim, diag=False, symperiod=None, dtype=complex):
+    # default standard deviation equals 1e-2
+    normal_init = jax.nn.initializers.normal()
+
+    L = hilbert.size
+    phys_dim = hilbert.local_size
+    diag = diag
+    dtype = dtype
+
+    # determine transformation from local states to indices
+    local_states = jax.numpy.array(hilbert.local_states)
+    loc_vals_spacing = jax.numpy.roll(local_states, -1)[0:-1] - local_states[0:-1]
+    if jax.numpy.max(loc_vals_spacing) == jax.numpy.min(loc_vals_spacing):
+        loc_vals_spacing = loc_vals_spacing[0]
+    else:
+        raise AssertionError(
+            "JaxMpsPeriodic can only be used with evenly spaced hilbert local values"
+        )
+    loc_vals_bias = jax.numpy.min(local_states)
+
+    # check whether graph is periodic chain
+    import networkx as _nx
+
+    edges = hilbert.graph.edges()
+    G = _nx.Graph()
+    G.add_edges_from(edges)
+
+    G_chain = _nx.Graph()
+    G_chain.add_edges_from([(i, (i + 1) % L) for i in range(L)])
+
+    if not _nx.algorithms.is_isomorphic(G, G_chain):
+        print(
+            "Warning: graph is not isomorphic to chain with periodic boundary conditions"
+        )
+
+    # determine shape of unit cell
+    if symperiod == None:
+        symperiod = L
+    if L % symperiod == 0 and symperiod > 0:
+        if diag:
+            unit_cell_shape = (symperiod, phys_dim, bond_dim)
+        else:
+            unit_cell_shape = (symperiod, phys_dim, bond_dim, bond_dim)
+    else:
+        raise AssertionError(
+            "The number of degrees of freedom of the Hilbert space needs to be a multiple of the period of the MPS"
+        )
+
+    # define diagonal tensors with correct unit cell shape
+    if diag:
+        iden_tensors = jax.numpy.ones((symperiod, phys_dim, bond_dim), dtype=dtype)
+    else:
+        iden_tensors = jax.numpy.repeat(
+            jax.numpy.eye(bond_dim, dtype=dtype)[jax.numpy.newaxis, :, :],
+            symperiod * phys_dim,
+            axis=0,
+        )
+        iden_tensors = iden_tensors.reshape(symperiod, phys_dim, bond_dim, bond_dim)
+
+    def init_fun(rng, input_shape):
+        random_tensors_real = normal_init(rng, unit_cell_shape)
+        if dtype == complex:
+            random_tensors_imag = normal_init(rng, unit_cell_shape)
```

Comment:
you have to split the rng, otherwise the real and imaginary part will be the same.

```python
    rng_re, rng_im = jax.random.split(rng)
    random_tensors_real = normal_init(rng_re, unit_cell_shape)
          if dtype == complex:
              random_tensors_imag = normal_init(rng_im, unit_cell_shape)
```

Reply:
good catch